`FileExistsError` error when `psyplot` is imported from multiple parallel processes

Hello!

First of all, thanks for this great package!!

#### Problem description

We (the [ESMValTool](https://github.com/ESMValGroup/ESMValTool) community) found an issue when importing `psyplot` from different parallel processes in a test environment, which leads to a race condition where the package finds that it has no config file and tries to create that, but two processes try that at the same time, resulting in a `FileExistsError: [Errno 17] File exists: '/Users/runner/.config/psyplot'`. Note that we only found this behavior for OSX.

#### Code Sample, a copy-pastable example if possible

An error on our CI machine using the import below is given [here](https://github.com/ESMValGroup/ESMValTool/runs/6470528634?check_suite_focus=true),

```python
from psyplot.config.logsetup import _get_home
```

but we also found the [same problem](https://github.com/ESMValGroup/ESMValTool/runs/6462593093?check_suite_focus=true#step:7:50) when using

```python
import psyplot.project as psy
```

Please note that due to the race condition, the error cannot be reproduced consistently.

#### Expected Output

No error and no race condition.

#### Full error message

<details>

```bash
 tests/unit/test_platform.py:4: in <module>
    from psyplot.config.logsetup import _get_home
../../../miniconda3/envs/esmvaltool/lib/python3.9/site-packages/psyplot/__init__.py:30: in <module>
    from psyplot.config.rcsetup import rcParams
../../../miniconda3/envs/esmvaltool/lib/python3.9/site-packages/psyplot/config/__init__.py:42: in <module>
    config_path = psyplot_fname()
../../../miniconda3/envs/esmvaltool/lib/python3.9/site-packages/psyplot/config/rcsetup.py:978: in psyplot_fname
    configdir = get_configdir()
../../../miniconda3/envs/esmvaltool/lib/python3.9/site-packages/psyplot/config/rcsetup.py:1028: in get_configdir
    os.makedirs(p)
../../../miniconda3/envs/esmvaltool/lib/python3.9/os.py:225: in makedirs
    mkdir(name, mode)
E   FileExistsError: [Errno 17] File exists: '/Users/runner/.config/psyplot'
```

</details>

#### Output of ``psyplot -aV``

<details>

```
psyplot:
  requirements:
    matplotlib: 3.5.2
    numpy: 1.22.3
    pandas: 1.4.2
    python: 3.8.13 | packaged by conda-forge
    xarray: 2022.3.0
  version: 1.4.2
```

</details>

#### NOTE
This is a bug report.

For requesting new features, use [this template](https://github.com/psyplot/psyplot/issues/new?template=new_feature.md&title=NEW+FEATURE:).

For changing existing features, use [this template](https://github.com/psyplot/psyplot/issues/new?template=change_feature.md&title=CHANGE+FEATURE:).

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -1,7 +1,7 @@
 version: 2.1
 
 orbs:
-  psyplot: psyplot/psyplot-ci-orb@1.5.31
+  psyplot: psyplot/psyplot-ci-orb@1.5.32
   mattermost-plugin-notify: nathanaelhoun/mattermost-plugin-notify@1.2.0
 
 executors:

--- a/psyplot/config/rcsetup.py
+++ b/psyplot/config/rcsetup.py
@@ -1025,7 +1025,7 @@ def get_configdir(name='psyplot', env_key='PSYPLOTCONFIGDIR'):
         p = os.path.join(h, '.' + name)
 
     if not os.path.exists(p):
-        os.makedirs(p)
+        os.makedirs(p, exist_ok=True)
     return p
 
 

--- a/psyplot/project.py
+++ b/psyplot/project.py
@@ -1163,7 +1163,7 @@ class Project(ArrayList):
         if pack and fname is not None:
             target_dir = os.path.dirname(fname)
             if not os.path.exists(target_dir):
-                os.makedirs(target_dir)
+                os.makedirs(target_dir, exist_ok=True)
 
             def tmp_it():
                 from tempfile import NamedTemporaryFile

